java.lang.SecurityException: One of RECEIVER_EXPORTED or RECEIVER_NOT_EXPORTED should be specified

[Newly introduced behavior changes in Android 14](https://developer.android.com/about/versions/14/behavior-changes-14) causes Maverick to crash the app.

<img width="888" alt="image" src="https://github.com/airbnb/mavericks/assets/2140323/8b3058fd-4d20-40ff-a68d-f4748f28c38b">

... 75 more
Caused by: java.lang.SecurityException: com.app: One of RECEIVER_EXPORTED or RECEIVER_NOT_EXPORTED should be specified when a receiver isn't being registered exclusively for system broadcasts
at android.os.Parcel.createExceptionOrNull([Parcel.java:3057](http://parcel.java:3057/))
at android.os.Parcel.createException([Parcel.java:3041](http://parcel.java:3041/))
at android.os.Parcel.readException([Parcel.java:3024](http://parcel.java:3024/))
at android.os.Parcel.readException([Parcel.java:2966](http://parcel.java:2966/))
at [android.app](http://android.app/).IActivityManager$Stub$Proxy.registerReceiverWithFeature([IActivityManager.java:5684](http://iactivitymanager.java:5684/))
at [android.app](http://android.app/).ContextImpl.registerReceiverInternal([ContextImpl.java:1852](http://contextimpl.java:1852/))
at [android.app](http://android.app/).ContextImpl.registerReceiver([ContextImpl.java:1792](http://contextimpl.java:1792/))
at [android.app](http://android.app/).ContextImpl.registerReceiver([ContextImpl.java:1780](http://contextimpl.java:1780/))
at android.content.ContextWrapper.registerReceiver([ContextWrapper.java:755](http://contextwrapper.java:755/))
at com.airbnb.mvrx.mocking.printer.MavericksPrintStateBroadcastReceiver.register(MavericksMockPrinter.kt:207)
at com.airbnb.mvrx.mocking.MockMavericksViewModelConfigFactory.buildConfig(MockableMavericksViewModelConfig.kt:273)
at com.airbnb.mvrx.MavericksViewModelConfigFactory.provideConfig$mvrx_release(MavericksViewModelConfigFactory.kt:67)
at com.airbnb.mvrx.MavericksViewModel.<init>(MavericksViewModel.kt:35)
at com.airbnb.mvrx.MavericksViewModel.<init>(MavericksViewModel.kt:25)

This happens due to the following line: https://github.com/airbnb/mavericks/blob/1318087ee6fea40d1d2e0b5a0b34f36c19a297dd/mvrx-mocking/src/main/kotlin/com/airbnb/mvrx/mocking/printer/MavericksMockPrinter.kt#L207

--- a/mvrx-mocking/src/main/kotlin/com/airbnb/mvrx/mocking/printer/MavericksMockPrinter.kt
+++ b/mvrx-mocking/src/main/kotlin/com/airbnb/mvrx/mocking/printer/MavericksMockPrinter.kt
@@ -206,11 +206,10 @@ internal abstract class MavericksPrintStateBroadcastReceiver : BroadcastReceiver
         check(!isRegistered) { "Already registered" }
         isRegistered = true
         if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU) {
-            context.registerReceiver(networkInfoReceiver, filter, Context.RECEIVER_NOT_EXPORTED)
+            context.registerReceiver(this, IntentFilter(ACTION_COPY_MAVERICKS_STATE), Context.RECEIVER_NOT_EXPORTED)
         } else {
-            context.registerReceiver(networkInfoReceiver, filter)
+            context.registerReceiver(this, IntentFilter(ACTION_COPY_MAVERICKS_STATE))
         }
-        context.registerReceiver(this, IntentFilter(ACTION_COPY_MAVERICKS_STATE))
     }
 
     fun unregister(context: Context) {

